src/libcore/hint.rs

```diff
@@ -81,10 +81,13 @@ pub fn spin_loop() {
     {
         #[cfg(target_arch = "aarch64")]
         {
+            // SAFETY: the `cfg` attr ensures that we only execute this on aarch64 targets.
```

Comment:
Why do we have this setup so strangely? Can we "unwrap" these inner cfg's?

AFAICT we should also be able to do the same for the x86/x86_64 cfg's, just gating on target_feature = "sse2" in the wrapper (or inlining that).

Reply:
This should probably be done in a separate PR?

Reply:
Yeah, just leaving feedback in case you're up for making that PR :)

Reply:
Seems like a low-effort low-hanging fruit. Yum!